x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/step_define_rule/index.tsx

```diff
@@ -446,37 +462,65 @@ const StepDefineRuleComponent: FC<StepDefineRuleProps> = ({
     ]
   );
 
+  const [{ queryBar }] = useFormData<DefineStepRule>({ form, watch: ['queryBar'] });
+  const areSuppressionFieldsDisabledBySequence =
+    isEqlRule(ruleType) &&
+    isEqlSequenceQuery(queryBar?.query?.query as string) &&
+    groupByFields.length === 0;
+
```

Comment:
When trying to clear invalid group by fields controller error disappears, but not configured fields.

Upon saving the rule, error appears again

https://github.com/elastic/kibana/assets/92328789/aa221189-51fa-4936-b836-0fe1fa44d842

Reply:
does not work when I tried to remove fields one by one

https://github.com/elastic/kibana/assets/92328789/c2fad415-7fa2-4382-9a6e-666dd3fc7835

Reply:
Happens on rule editing, not rule creation

Reply:
Can you please provide steps to reproduce this? This appears to be working differently for me: both the clear and individual removal of fields works (on create and edit), and once all fields are removed the suppression fields become disabled.

Reply:
So, cypress tests started failing in 7aa52b8bde8a829ae8e2cedd357b051ce4549767, and when I reverted that to attempt to fix tests, I was able to reproduce the behavior above. So one implementation fails in cypress, and the other causes a UI bug.... 🤔